### ISSUE TITLE
Inconsistent behavior of boolean and option in strict mode

### ISSUE THREAD

**Issue by jhiode** [yargs/yargs#1694]
I found an inconsistent behavior of `.boolean(key)` and `.option(key, { boolean: true })`.

Here is an example:
```js
const yargs = require('yargs'); // v15.4.1

yargs
  .command({
    command: 'test',
    builder: (argv) => {
      return argv.boolean('b1').option('b2', { boolean: true });
    },
    handler: console.log,
  })
  .strict()
  .parse();
```

As far as I understand the documentation, these two APIs should work the same, but only option b2 is actually working, option b1 is simply ignored:

```
$ node index.js test --help
index.js test

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --b2                                                                 [boolean]
```

Using it results in an error:
```
$ node index.js test --b1
index.js test

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --b2                                                                 [boolean]

Unknown argument: b1
```

But it works in non-strict mode though:
```
$ node index.js test --b1
{ _: [ 'test' ], b1: true, '$0': 'index.js' }
```

It seems that the call to `self.describe` is the reason for the different behavior.
https://github.com/yargs/yargs/blob/0b519a4672c8493d72838292ed0d60b63b88f33e/lib/yargs.ts#L843-L844

**Comment by jly36963** [comment]
@jhiode 
I believe [my fix](https://github.com/yargs/yargs/pull/1996) has been released with 17.3.1 -- can you confirm that it solves the issue you reported?

**Comment by shadowspawn** [comment]
Thanks for the report. This was likely fixed by #1996 by @jly36963.

This issue has not had any activity in over six months. It isn't likely to get acted on due to this report.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you  for your contributions.
